### PR TITLE
Added 100x faster vectorized version

### PR DESCRIPTION
**Pull request by parejkoj**
Would be ~500x faster if we didn't have to do the footprint loop.

The if branch between vector and non-vectorized versions (and associated split in code logic) is unpleasant, but we'll have to live with it until we finalize the discussion here: https://community.lsst.org/t/memory-contiguity-and-python-execution-speed/884/12

### CODE REVIEW

File: python/lsst/meas/algorithms/astrometrySourceSelector.py [hunk 1]
```diff
@@ -68,16 +72,24 @@ def selectSources(self, sourceCat, matches=None):
...
+                if self._isGood(source) and not self._isBad(source):
+                    result.append(source)
         return Struct(sourceCat=result)
 
     def _getSchemaKeys(self, schema):
         """Extract and save the necessary keys from schema with asKey."""
         self.parentKey = schema["parent"].asKey()
         self.nChildKey = schema["deblend_nChild"].asKey()
+        self.centroidXKey = schema["slot_Centroid_x"].asKey()
+        self.centroidYKey = schema["slot_Centroid_y"].asKey()
```
Comment (isullivan) [inline python/lsst/meas/algorithms/astrometrySourceSelector.py]: I thought the field was "slot_Centroid" and it returned a tuple of the (y, x) centroid location. Does this also work?

Comment (parejkoj) [inline python/lsst/meas/algorithms/astrometrySourceSelector.py]: That works for individual sources, but I need the x/y centroids for the vectorized np.isfinite() test: there isn't an equivalent SourceCatalog thing to get a vector of tuples.

File: python/lsst/meas/algorithms/astrometrySourceSelector.py [hunk 2]
```diff
@@ -68,16 +72,24 @@ def selectSources(self, sourceCat, matches=None):
...
-                result.append(source)
+        if sourceCat.isContiguous():
+            bad = reduce(lambda x, y: np.logical_or(x, sourceCat.get(y)), self.config.badFlags, False)
+            good = self._isGood_vector(sourceCat)
+            result = sourceCat[good & ~bad]
+        else:
+            result = table.SourceCatalog(sourceCat.table)
+            for i, source in enumerate(sourceCat):
+                if self._isGood(source) and not self._isBad(source):
+                    result.append(source)
```
Comment (isullivan) [inline python/lsst/meas/algorithms/astrometrySourceSelector.py]: I don't like the code duplication of having vectorized and non-vectorized versions of every function, but I suppose that can't be helped.

File: python/lsst/meas/algorithms/astrometrySourceSelector.py (hunk 1 above)
Comment (TallJimbo) [inline python/lsst/meas/algorithms/astrometrySourceSelector.py]: Small correction: when used as a single key `slot_Centroid` will return `afw.geom.Point2D`, which is indexable as `(x,y)`, not `(y,x)`.  The usage here is completely valid too, of course.

File: python/lsst/meas/algorithms/astrometrySourceSelector.py (hunk 2 above)
Comment (parejkoj) [inline python/lsst/meas/algorithms/astrometrySourceSelector.py]: I agree, but until we have a generic solution, I'd rather not break non-contiguous functionality.

**Pull request merged**